RecoTauTag/HLTProducers/src/TauTagFilter.cc

```diff
@@ -0,0 +1,171 @@
+        matchingdR2_(std::pow(cfg.getParameter<double>("matchingdR"), 2)) {
+    if (cfg.getParameter<bool>("matchWithSeeds") && cfg.getParameter<double>("matchingdR") < 0)
+      edm::LogWarning("TauTagFilter") << "Matching with seeds is disabled because matchingdR < 0";
+
+    extractMomenta();  // checking that all seed types are supported
+  }
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    makeHLTFilterDescription(desc);
+    desc.add<int>("nExpected", 2)->setComment("number of expected taus per event");
+    desc.add<edm::InputTag>("taus", edm::InputTag(""))->setComment("input collection of taus");
+    desc.add<edm::InputTag>("tauTags", edm::InputTag(""))->setComment("input collection of tau tagger scores");
+    desc.add<edm::InputTag>("tauPtCorr", edm::InputTag(""))
+        ->setComment("input collection of multiplicative tau pt corrections");
+    desc.add<edm::InputTag>("seeds", edm::InputTag(""))->setComment("input collection of seeds");
+    desc.add<std::vector<int>>("seedTypes",
+                               {trigger::TriggerL1Tau, trigger::TriggerL1Jet, trigger::TriggerTau, trigger::TriggerJet})
+        ->setComment("list of seed object types");
+    desc.add<std::string>("selection", "0")->setComment("selection formula");
+    desc.add<double>("minPt", 20)->setComment("minimal tau pt");
+    desc.add<double>("maxEta", 2.5)->setComment("maximal tau abs(eta)");
+    desc.add<bool>("usePtCorr", false)->setComment("use multiplicative tau pt corrections");
+    desc.add<bool>("matchWithSeeds", false)->setComment("apply match with seeds");
+    desc.add<double>("matchingdR", 0.5)->setComment("deltaR for matching with seeds");
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+  bool hltFilter(edm::Event& event,
+                 const edm::EventSetup& eventsetup,
+                 trigger::TriggerFilterObjectWithRefs& filterproduct) const override {
+    if (saveTags())
+      filterproduct.addCollectionTag(tausSrc_);
+
+    int nTauPassed = 0;
+
+    const auto tausHandle = event.getHandle(tausToken_);
+    const auto& taus = *tausHandle;
+
+    const std::vector<LorentzVectorM> seed_p4s = extractMomenta(&event);
+    auto hasMatch = [&](const LorentzVectorM& p4) {
+      for (const auto& seed_p4 : seed_p4s) {
+        if (reco::deltaR2(p4, seed_p4) < matchingdR2_)
+          return true;
+      }
+      return false;
+    };
+
+    const auto& tauTags = event.get(tauTagsToken_);
+    const TauTagCollection* tauPtCorr = nullptr;
+    if (usePtCorr_)
+      tauPtCorr = &event.get(tauPtCorrToken_);
+
+    if (taus.size() != tauTags.size())
+      throw cms::Exception("Inconsistent Data", "PNetTauTagFilter::hltFilter") << "taus.size() != tauTags.size()";
+    if (usePtCorr_ && taus.size() != tauPtCorr->size())
+      throw cms::Exception("Inconsistent Data", "PNetTauTagFilter::hltFilter") << "taus.size() != tauPtCorr.size()";
```

Comment:
```suggestion
      throw cms::Exception("Inconsistent Data", "TauTagFilter::hltFilter") << "taus.size() != tauPtCorr.size()";
```
to match the class name.